src/otx/core/model/entity/base.py

```diff
@@ -186,13 +187,15 @@ def export(
             base_name: (str): base name for the exported model file. Extension is defined by the target export format
             export_format (OTXExportFormatType): format of the output model
             precision (OTXExportPrecisionType): precision of the output model
+            test_pipeline (list[dict] | None): test dataset pipeline. necessary for mmdeploy.
+
         Returns:
             Path: path to the exported model.
         """
-        return self._exporter.export(self.model, output_dir, base_name, export_format, precision)
+        exporter = self._get_exporter(test_pipeline)
+        return exporter.export(self.model, output_dir, base_name, export_format, precision)
 
-    @property
-    def _exporter(self) -> OTXModelExporter:
+    def _get_exporter(self, test_pipeline: list[dict] | None) -> OTXModelExporter:
```

Comment:
I'd opt for adding `test_pipeline` via `_export_parameters` (it has to be a class attribute then) rather than damaging the interface. @vinnamkim what do you think? Perhaps, new class attribute will affect snapshot structure, then that way won't work

Reply:
I also agree with you but it's hard to find elegant way. Although I thought about various things, it seems strange for models to have a test dataset pipeline which is dataset related things. How do you think about it?

Reply:
> I'd opt for adding `test_pipeline` via `_export_parameters` (it has to be a class attribute then) rather than damaging the interface. @vinnamkim what do you think? Perhaps, new class attribute will affect snapshot structure, then that way won't work

Yea, it looks good to me. Please keep in mind that everything which is needed for the model exportation can be found from `_export_parameters` and please do not make a different source to avoid searching multiple places in the code base like OTX 1.x.

Shouldn't `test_pipeline` be configured by the model configuration? If so, I don't see the reason why it's exposed and injected from outside.

Reply:
yes, it's impossible to get `test_pipeline` from model config.  so if we want to gather all info in single place, I think there are two options. One is passing `test_pipeline` to `OTXModel.__init__()`(maybe `OTXDetectionModel.__init__()`?) as argument. But is it ok? `test_pipeline` is required only when exporting det and i-seg models. So we should add arguments which is necessary only in limited cases. Moreover, if user decides to create model outside `Engine`, then user should pass `test_pipeline` by himself when creating model instance only for limited case.  Other option is setting `test_pipeline` after instnace is created. but I think it's bad idea. It's becase there is no hint that `test_pipeline` need to be set after instance is created for exporting det or i-seg models. Developer needs to inspect export code to infer it. These things are my concerns. If you have other ides to avoid it, pelase let me know :) or even with this downside, if you think it's better than current design, then I'll follow it.

Reply:
I heard the idea from Vinnam in offline and trying to apply it.

Reply:
done at https://github.com/openvinotoolkit/training_extensions/pull/2806/commits/a3a7ec55961379f670af4cdbd8d852c4c0875772.
Vinnam's idea is to make fake `test_pipeline` referring input_size. I checked it works (IR file is exactly same).